browse/controllers/archive_page/__init__.py

```diff
@@ -0,0 +1,133 @@
+
+    subsumed_by = ARCHIVES_SUBSUMED.get(archive_id, None)
+    if subsumed_by:
+        data["subsumed_id"] = archive_id
+        data["subsumed_category"] = CATEGORIES.get(archive_id, {})
+        data["subsumed_by"] = subsumed_by
+        subsuming_category = CATEGORIES.get(subsumed_by, {})
+        data["subsuming_category"] = subsuming_category
+        archive_id = subsuming_category.get("in_archive", None)
+        archive = ARCHIVES.get(archive_id, None)
+
+    years = years_operating(archive)
+
+    data["years"] = years
+    data["months"] = MONTHS
+    data["days"] = DAYS
+    
+    data["archive_id"] = archive_id
+    data["archive"] = archive
+    data["list_form"] = ByMonthForm(archive_id, archive, years)
+    data["stats_by_year"] = stats_by_year(archive_id, archive, years)
+    data["category_list"] = category_list(archive_id)
+
+    data["catchup_to"] = datetime.date.today() - datetime.timedelta(days=7)
+    
+    data["template"] = "archive/single_archive.html"
+    return data, status.HTTP_200_OK, response_headers  # type: ignore
+
+
+def archive_index(archive_id: str, status: int) -> Response:
+    """Landing page for when there is no archive specified."""
+    data: Dict[str, Any] = {}
+    data["bad_archive"] = archive_id
+
+    archives = [
+        (id, ARCHIVES[id]["name"])
+        for id in ARCHIVES.keys()
+        if id not in ARCHIVES_SUBSUMED and not id.startswith("test")
+    ]
+    archives.sort(key=lambda tpl: tpl[0])
+    data["archives"] = archives
+
+    defunct = [
+        (id, ARCHIVES[id]["name"], ARCHIVES_SUBSUMED.get(id, ""))
+        for id in ARCHIVES.keys()
+        if "end_date" in ARCHIVES[id]
+    ]
+    defunct.sort(key=lambda tpl: tpl[0])
+    data["defunct"] = defunct
+    
+    data["template"] = "archive/archive_list_all.html"
+    return data, status, {}  # type: ignore
+
+
+def subsumed_msg(archive: Dict[str, str], subsumed_by: str) -> Dict[str, str]:
+    """Adds information about subsuming categories and archives."""
+    sb = CATEGORIES.get(subsumed_by, {"name": "unknown category"})
+    sa = ARCHIVES.get(sb.get("in_archive", None), {"name": "unknown archive"})
+
+    return {"subsumed_by_cat": sb, "subsumed_by_arch": sa}
+
+
+def category_list(archive_id: str) -> List[Dict[str, str]]:
+    """Retunrs categories for archive."""
+    cats = []
+    for cat_id in CATEGORIES:
+        cat = CATEGORIES[cat_id]
+        if(cat.get("in_archive", "yuck") == archive_id
+           and cat.get("is_active", True)):
+            cats.append({"id": cat_id,
+                         "name": cat.get("name", ""),
+                         "description": cat.get("description", "")})
+
+    cats.sort(key=lambda x: x["name"])
+    return cats
+
+
+def _write_expires_header(response_headers: Dict[str, Any]) -> None:
+    """Writes an expires header for the response."""
+    response_headers["Expires"] = abs_expires_header()[1]
+
+    
+DAYS = ["{:0>2d}".format(i) for i in range(1, 32)]
+
+MONTHS = [
```

Comment:
Any reason not to define this programmatically, using https://docs.python.org/3/library/calendar.html#calendar.month_abbr ?

```python
MONTHS = [(m, '%s (%s)' % (m, calendar.month_abbr[i])) for i, m 
          in enumerate(map(lambda o: str(o).zfill(2), range(0, 13) ))][1:]
```

Reply:
I would not choose to define it programmatically because the literal syntax seems simpler and more clear as to the final shape of the data.

Reply:
That's fair